smartlearner/views.py

```diff
@@ -51,3 +51,39 @@ def mean(self):
     @property
     def stderror(self):
         return ItemGetter(self, attribute=1)
+
+
+class ReconstructionError(View):
+    def __init__(self, predict_fct, dataset):
+        super().__init__()
+
+        batch_size = 1024  # Internal buffer
+        self.nb_batches = int(np.ceil(len(dataset) / batch_size))
+
+        input = dataset.symb_inputs
+        target = dataset.symb_targets
+        reconstruction_errors = T.sqr(predict_fct(input) - target).mean()
```

Comment:
Why would I want to do that? It doesn't make sense.

Adam Salvail
Doctorant en informatique
Université de Sherbrooke
819 821-8000 (62038)

On 19/08/15 04:26 PM, Marc-Alexandre Côté wrote:
>
> In smartlearner/views.py 
> <https://github.com/SMART-Lab/smartlearner/pull/43#discussion_r37463452>:
>
> > @@ -51,3 +51,39 @@ def mean(self):
> >      @property
> >      def stderror(self):
> >          return ItemGetter(self, attribute=1)
> > +
> > +
> > +class ReconstructionError(View):
> > +    def __init__(self, predict_fct, dataset):
> > +        super().__init__()
> > +
> > +        batch_size = 1024  # Internal buffer
> > +        self.nb_batches = int(np.ceil(len(dataset) / batch_size))
> > +
> > +        input = dataset.symb_inputs
> > +        target = dataset.symb_targets
> > +        reconstruction_errors = T.sqr(predict_fct(input) - target).mean()
>
> Ah, you take the mean here. I think you want to take the mean along 
> axis=1 and not across the examples of the batch (this one will be done 
> below).
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/SMART-Lab/smartlearner/pull/43/files#r37463452>.Web Bug 
> from 
> https://github.com/notifications/beacon/AEJelE30FHfKNLAujP4KAkmhC7EQnjCoks5opN33gaJpZM4FuntY.gif
>